optuna/integration/allennlp.py

```diff
@@ -1,501 +1,3 @@
-        return None
-
-    return pruner(**pruner_params)
-
-
-def _get_environment_variables_for_trial() -> Dict[str, Optional[str]]:
-    return {
-        "study_name": os.getenv(_STUDY_NAME),
-        "trial_id": os.getenv(_TRIAL_ID),
-        "storage": os.getenv(_STORAGE_NAME),
-        "monitor": os.getenv(_MONITOR),
-    }
-
-
-def _get_environment_variables_for_pruner() -> Dict[str, Optional[Union[str, int, float, bool]]]:
-    keys = os.getenv(_PRUNER_KEYS)
-
-    # keys would be empty when `_PRUNER_CLASS` is `NopPruner`
-    if keys is None or keys == "":
-        return {}
-
-    kwargs = {}
-    for key in keys.split(","):
-        key_without_prefix = key.replace("{}_".format(_PREFIX), "")
-        value = os.getenv(key)
-        if value is None:
-            raise ValueError(f"{key} is not found in environment variables.")
-        kwargs[key_without_prefix] = eval(value)
-
-    return kwargs
-
-
-def _fetch_pruner_config(trial: optuna.Trial) -> Dict[str, Any]:
-    pruner = trial.study.pruner
-    kwargs: Dict[str, Any] = {}
-
-    if isinstance(pruner, optuna.pruners.HyperbandPruner):
-        kwargs["min_resource"] = pruner._min_resource
-        kwargs["max_resource"] = pruner._max_resource
-        kwargs["reduction_factor"] = pruner._reduction_factor
-
-    elif isinstance(pruner, optuna.pruners.MedianPruner):
-        kwargs["n_startup_trials"] = pruner._n_startup_trials
-        kwargs["n_warmup_steps"] = pruner._n_warmup_steps
-        kwargs["interval_steps"] = pruner._interval_steps
-
-    elif isinstance(pruner, optuna.pruners.PercentilePruner):
-        kwargs["percentile"] = pruner._percentile
-        kwargs["n_startup_trials"] = pruner._n_startup_trials
-        kwargs["n_warmup_steps"] = pruner._n_warmup_steps
-        kwargs["interval_steps"] = pruner._interval_steps
-
-    elif isinstance(pruner, optuna.pruners.SuccessiveHalvingPruner):
-        kwargs["min_resource"] = pruner._min_resource
-        kwargs["reduction_factor"] = pruner._reduction_factor
-        kwargs["min_early_stopping_rate"] = pruner._min_early_stopping_rate
-
-    elif isinstance(pruner, optuna.pruners.ThresholdPruner):
-        kwargs["lower"] = pruner._lower
-        kwargs["upper"] = pruner._upper
-        kwargs["n_warmup_steps"] = pruner._n_warmup_steps
-        kwargs["interval_steps"] = pruner._interval_steps
-    elif isinstance(pruner, optuna.pruners.NopPruner):
-        pass
-    else:
-        raise ValueError("Unsupported pruner is specified: {}".format(type(pruner)))
-
-    return kwargs
-
-
-def _is_encodable(value: str) -> bool:
-    # https://github.com/allenai/allennlp/blob/master/allennlp/common/params.py#L77-L85
-    return (value == "") or (value.encode("utf-8", "ignore") != b"")
-
-
-def _environment_variables() -> Dict[str, str]:
-    return {key: value for key, value in os.environ.items() if _is_encodable(value)}
-
-
-def dump_best_config(input_config_file: str, output_config_file: str, study: optuna.Study) -> None:
```

Comment:
[Note] Moved to `_allennlp/dump_best_config.py` and `_allennlp/pruner.py`. Should remove the one or the other.